connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryListener.scala

```diff
@@ -136,6 +151,99 @@ object StreamingQueryListener extends Serializable {
     private def jsonValue: JValue = JObject("progress" -> progress.jsonValue)
   }
 
+  private[spark] object QueryProgressEvent {
+    private implicit val formats: DefaultFormats = DefaultFormats
+
+    def fromJson(input: String): QueryProgressEvent = {
```

Comment:
For these types of methods, do you think we can use something like this?
https://github.com/apache/spark/blob/ae5f7619a057de1eeba3565f302767b9262ed61c/sql/core/src/main/scala/org/apache/spark/sql/streaming/progress.scala#L181-L194

Also the change should happen in both listener files to sync up I think, one is this
connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryListener.scala
another is this
sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryListener.scala

Reply:
Thanks yeah I missed that, will update the core one.

The default `fromJson()` doesn't seem to work in real test since it was not able to handle null value, I remember I tried this way at first but didn't go well. I can double check to see if the null issue can be fixed